### PR TITLE
Added support so that smile can use hats

### PR DESCRIPTION
**Pull request by zHonys**
## About the PR

Made that the smile pet can use hats;
It also sets up use for mask and gas tank for smile, but since i don't have new sprites for these cases, I will leave the option open for someone else.

## Why / Balance

Couldn't affect game balance. 
It was changed to make possible that smile uses a hat

## Technical details

Moved Resources/Textures/Mobs/Pets/smile.rsi/ to .../Pets/smile/smile.rsi/
Changed .../Resources/Prototypes/Entities/Mobs/NPCs/pets.yml
Added .../smile_inventory_template.yml in .../Resources/Prototypes/inventoryTemplates 
Added dir smile_displacement.rsi inside Resources/Textures/Mobs/Pets/smile/ Added smile_displacement.rsi/meta.json
Added .../smile/smile_displacement.rsi/head.png

## Media
Alien plushie hat
Beaver hat
image

## Requirements

- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

Won't have braking changes

**Changelog**

:cl:
- add: Smile can now use hats

### CODE REVIEW

**Comment by beck-thompson** [comment]
You could probably add a change log if you want! This kind of thing is fun

**Comment by ArtisticRoomba** [comment]
definitely also add some media so we can see what she looks like with the drip on

**Comment by MossyGreySlope** [comment]
I tried putting the magician's tophat on Smile, and it seems like the sprite is broken. Other hats all seem to fail as well.

https://github.com/user-attachments/assets/c23e78c2-0874-4717-86da-bdd8ae9aeff0

**Comment by zHonys** [comment]
> definitely also add some media so we can see what she looks like with the drip on

I did add a media and change log, don't know why it didn't appear. I'll try editing it

**Comment by zHonys** [comment]
> I tried putting the magician's tophat on Smile, and it seems like the sprite is broken. Other hats all seem to fail as well.
> smile.mp4

I think i did something wrong on the PR then, i'll try to fix it. Sorry for this, first time using the PR feature on github.

**Comment by zHonys** [comment]
> I tried putting the magician's tophat on Smile, and it seems like the sprite is broken. Other hats all seem to fail as well.
> smile.mp4

it should be fixed, i moved the displacement sprite path and didn't commit it. Should be all good for now!

**Comment by MossyGreySlope** [comment]
Seems fixed. Nice work.

https://github.com/user-attachments/assets/68d2e742-9c1b-494d-bb27-bda06faeb3c6

File: Resources/Prototypes/InventoryTemplates/smile_inventory_template.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+- type: inventoryTemplate
+  id: smile
+  slots:
+  - name: head
+    slotTexture: head
+    slotFlags: HEAD
+    uiWindowPos: 0,1
+    strippingWindowPos: 0,0
+    displayName: Head
+  # Commented since that would need specific art for smile, at least for the breathing mask and gas tanks
```
Comment (TheShuEd) [inline Resources/Prototypes/InventoryTemplates/smile_inventory_template.yml:10]: Just remove comments

File: Resources/Prototypes/Entities/Mobs/NPCs/pets.yml [hunk 2]
```diff
@@ -722,6 +722,17 @@
     - map: [ "enum.DamageStateVisualLayers.BaseUnshaded" ]
       state: aslime-_3
       shader: unshaded
+    #- map: [ "mask" ] breathing mask sprites need new versions for smile
```
Comment (TheShuEd) [inline Resources/Prototypes/Entities/Mobs/NPCs/pets.yml:725]: Remove

File: Resources/Prototypes/Entities/Mobs/NPCs/pets.yml [hunk 3]
```diff
@@ -767,7 +778,7 @@
         Caustic: 1
   - type: MultiHandedItem
   - type: Item
-    sprite: Mobs/Pets/smile.rsi
+    sprite: Mobs/Pets/smile/smile.rsi
```
Comment (TheShuEd) [inline Resources/Prototypes/Entities/Mobs/NPCs/pets.yml:781]:
```suggestion
    sprite: Mobs/Pets/Smile/smile.rsi
```

folder naming convention

**Review by TheShuEd — CHANGES_REQUESTED** [review]

**Comment by zHonys** [comment]
I was hoping someone could use these to add the mask and gas tank to smile as well, but I'll remove it then

File: Resources/Prototypes/InventoryTemplates/smile_inventory_template.yml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+- type: inventoryTemplate
```
Comment (TheShuEd) [inline Resources/Prototypes/InventoryTemplates/smile_inventory_template.yml:1]: Check another inventory templates
100% you can find that one, looking exactly same (with only head slot)

Comment (zHonys) [reply Resources/Prototypes/InventoryTemplates/smile_inventory_template.yml]: It does, forgot about it when removing the comments

**Review by TheShuEd — APPROVED** [review]

**Review by slarticodefast — APPROVED** [review]
grafik
grafik
grafik
You can now give smile cat ears.
Based first PR. Thank you for your contribution.

**Pull request merged**